### ISSUE TITLE
Cannot get contributed pack: 'sys' has no attribute 'exc_type' 

### ISSUE THREAD

**Issue by p-a-s-c-a-l** [nextgis/quickmapservices#180]
The following exception is thrown when trying to get the contributed pack:

AttributeError: module 'sys' has no attribute 'exc_type' 
```
Traceback (most recent call last):
  File "C:/Users/pd/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\quick_map_services\settings_dialog.py", line 127, in get_contrib
    ExtraSources().load_contrib_pack()
  File "C:/Users/pd/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\quick_map_services\extra_sources.py", line 87, in load_contrib_pack
    self._download_file(CONTRIBUTE_ZIP_DIRECT_URL, zip_file_path)
  File "C:/Users/pd/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\quick_map_services\extra_sources.py", line 116, in _download_file
    reply = self.__sync_request(url)
  File "C:/Users/pd/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\quick_map_services\extra_sources.py", line 145, in __sync_request
    raise Exception(error)
Exception: 99

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:/Users/pd/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\quick_map_services\settings_dialog.py", line 135, in get_contrib
    error_message = self.tr('Error on getting contrib pack: %s %s') % (sys.exc_type, sys.exc_value)
AttributeError: module 'sys' has no attribute 'exc_type'
```

**Python-Version: 3.7.0 (v3.7.0:1bf9cc5093, Jun 27 2018, 04:59:51) [MSC v.1914 64 bit (AMD64)]**
**QGIS-Version: 3.4.1-Madeira Madeira, 383851c597**

**Comment by p-a-s-c-a-l** [comment]
Tried a re-install of the extension in qGIS 3.4.2 and encountered the following error:

**Could not load extension 'quick_map_services' due to error calling classFactory() method**

TypeError: 'QVariant' object is not subscriptable 
Traceback (most recent call last):
  File "C:/PROGRA~1/QGIS3~1.4/apps/qgis/./python\qgis\utils.py", line 335, in startPlugin
    plugins[packageName] = package.classFactory(iface)
  File "C:/Users/pd/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\quick_map_services\__init__.py", line 35, in classFactory
    return QuickMapServices(iface)
  File "C:/Users/pd/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\quick_map_services\quick_map_services.py", line 72, in __init__
    self.locale = Locale.get_locale()
  File "C:/Users/pd/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\quick_map_services\plugin_locale.py", line 30, in get_locale
    return QSettings().value('locale/userLocale')[0:2]
TypeError: 'QVariant' object is not subscriptable


**Python-Version: 3.7.0 (v3.7.0:1bf9cc5093, Jun 27 2018, 04:59:51) [MSC v.1914 64 bit (AMD64)]** 
**QGIS-Version: 3.4.2-Madeira Madeira, 22034aa070**

**Comment by tatva181** [comment]
Any luck?? @p-a-s-c-a-l

**Comment by Lallimost** [comment]
I have the same problem:

AttributeError: module 'sys' has no attribute 'exc_type' 
Traceback (most recent call last):
  File "C:/Users/mostarda/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\quick_map_services\settings_dialog.py", line 127, in get_contrib
    ExtraSources().load_contrib_pack()
  File "C:/Users/mostarda/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\quick_map_services\extra_sources.py", line 87, in load_contrib_pack
    self._download_file(CONTRIBUTE_ZIP_DIRECT_URL, zip_file_path)
  File "C:/Users/mostarda/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\quick_map_services\extra_sources.py", line 116, in _download_file
    reply = self.__sync_request(url)
  File "C:/Users/mostarda/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\quick_map_services\extra_sources.py", line 145, in __sync_request
    raise Exception(error)
Exception: 2

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:/Users/mostarda/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\quick_map_services\settings_dialog.py", line 135, in get_contrib
    error_message = self.tr('Error on getting contrib pack: %s %s') % (sys.exc_type, sys.exc_value)
AttributeError: module 'sys' has no attribute 'exc_type'


Versione Python: 3.7.0 (v3.7.0:1bf9cc5093, Jun 27 2018, 04:59:51) [MSC v.1914 64 bit (AMD64)] 
Versione di QGIS: 3.4.4-Madeira Madeira, f6ddc62fdb

**Comment by jonnyforestGIS** [comment]
Hi guys,

**Windows 10 Home 64-bits**
Tested in QGIS 3.4.5 Madeira LTR - Not OK
Tested in QGIS 3.6 Noosa - Not OK

This is broken for Windows QGIS Installer.

**Linux**
Kubuntu 64-bits - OK

Can be a python problem?

**Comment by datapolitan** [comment]
> Beside of this error as a workaround, you can open qms toolbox and access to the services and add the maps.

Hi @jonnyforestGIS, I'm having the same issue with Windows 7 machines and can't find any reference to the qms toolbox you reference. Can you please provide a link to some documentation or a tutorial on how to implement this workaround to "access to the services and add the maps"?

**Comment by kmezhoud** [comment]
Hi,
I tried to install **QuickMapServices** in OSX 10.15.2 through plugin manager setting. 
It returns a log error message:
`unavailable  https://plugins.qgis.org/plugins/plugins.xml`
I have to load the zip and install it manually.

During default setting of **QuickMapServices** by: 

`Web > QuickMapServices Settings > More services > Get contributed pack`

I got this error message:

```
AttributeError: module 'sys' has no attribute 'exc_type'
    Traceback (most recent call last):
      File "/Users/Mezhoud/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/quick_map_services/settings_dialog.py", line 127, in get_contrib
        ExtraSources().load_contrib_pack()
      File "/Users/Mezhoud/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/quick_map_services/extra_sources.py", line 87, in load_contrib_pack
        self._download_file(CONTRIBUTE_ZIP_DIRECT_URL, zip_file_path)
      File "/Users/Mezhoud/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/quick_map_services/extra_sources.py", line 116, in _download_file
        reply = self.__sync_request(url)
      File "/Users/Mezhoud/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/quick_map_services/extra_sources.py", line 145, in __sync_request
        raise Exception(error)
    Exception: 99
     
    During handling of the above exception, another exception occurred:
     
    Traceback (most recent call last):
      File "/Users/Mezhoud/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/quick_map_services/settings_dialog.py", line 135, in get_contrib
        error_message = self.tr('Error on getting contrib pack: %s %s') % (sys.exc_type, sys.exc_value)
    AttributeError: module 'sys' has no attribute 'exc_type'
     
     
    Python version: 3.7.3 (default, Mar 27 2019, 09:23:15) [Clang 10.0.1 (clang-1001.0.46.3)] 
    QGIS version: 3.10.2-A Coruña A Coruña, 616ad4531b 
     
    Python Path:
    /Applications/QGIS3.10.app/Contents/MacOS/../Resources/python
    /Users/Mezhoud/Library/Application Support/QGIS/QGIS3/profiles/default/python
    /Users/Mezhoud/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins
    /Applications/QGIS3.10.app/Contents/MacOS/../Resources/python/plugins
    /Applications/QGIS3.10.app/Contents/Resources/python
    /Applications/QGIS3.10.app/Contents/Frameworks/Python.framework/Versions/Current/lib/python3.7/lib-dynload
    /Applications/QGIS3.10.app/Contents/Frameworks/Python.framework/Versions/Current/lib/python3.7/site-packages
    /Users/Mezhoud/Library/Python/3.7/lib/python/site-packages
    /Applications/QGIS3.10.app/Contents/Frameworks/Python.framework/Versions/Current/lib/python3.7
    /Applications/QGIS3.10.app/Contents/Frameworks/Python.framework/Versions/Current/lib/python37.zip
    /Applications/QGIS3.10.app/Contents/Frameworks/Python.framework/Versions/Current/lib/python3.7/site-packages/geos
    /Users/Mezhoud/Library/Application Support/QGIS/QGIS3/profiles/default/python 

```

**Comment by aledalla** [comment]
The problem still occur in the last versions of QGIS. 
I have some machines and I get the error only on Windows 7; in Windows 10 it seems work.
I Can't get ride of this..
Someone of nextgis developers could help?
P.S. Sorry for my bad english

**Comment by tatva181** [comment]
Hi Aledalla,

We have faced such issue in the beginning of application setup and it was coming due to proxy settings, in case if the proxy setting are not configured properly then it will not allow us to download the contributed pack.



Best Regards,  

JAYDEEP CHAVDA  

Email Id:  jaydeep.chavda@tatvasoft.com

Skype Id:  tatva181

**Comment by aledalla** [comment]
Thank you tatva181,

As you suggested, the problem is related to proxy.
I tried to not proxy the connection and everithing was fine, I will find out what is wrong on my corporate proxy; maybe you can give me some hints about how the requests are made when downloading the contribute pack.

If anyone still have this problem I suggest to investigate on that.

**Comment by 514ckw4r3** [comment]
Fixed in version 0.19.24. The fix will be available for download in the next release.

**Comment by CriticalWill** [comment]
QGIS 3.28.8 
receiving same error. Windows 10 OS, no proxy

>
